### DISCUSSION TITLE
'Problems' tab is not updated properly after empty diagnostics.

### DISCUSSION THREAD

**Discussion by aghayanlena** [eclipse-theia/theia#13424] [Q&A]
Hi, 

I have a Language Server VSCode extension installed on my custom Theia application (v1.44). 
After each didChange notification, my LSP server pushes the diagnostics information(textDocument/publishDiagnostics) back to the client multiple times (we don't want the user to wait until full diagnostics is ready as it can take time). As I understood, the client should override the diagnostics for each URI after receiving a diagnostics notification from the server (from LSP spec there should not be merging on the client side).  

After installing my vsix on Theia the Problems tab is not updated properly when the client receives **empty** diagnostics. The errors count on the Problems tab properly gets updated, and the error squiggle of the previous error message disappears. However, I still see the previous error message on the Problems tab. 
I have not seen this behavior on VSCode. 

I don't see this behavior when sending only one pushDiagnostics notification from the server after each didChange for each URI.  For my case, it will be better to send the diagnostic with the portions. 

Same behavior with simple lps sample after sending multiple publishDiagnostics after each didChange (https://github.com/microsoft/vscode-extension-samples/tree/main/lsp-sample)
Same behavior with the latest Theia IDE (1.46).

Is this an issue in Theia?  If yes, is there any workaround that I can use until it gets fixed?

**Comment by JonasHelming** [comment]
This sounds like a bug. If you have steps to reproduce, could you create a abug report, ideally with the extensions required to reproduce this.
